### PR TITLE
Fix #4758: Revert "New phase for entering annotations"

### LINKED ISSUE #4758: Cyclic error involving annotation
In a file `Test1.scala`:
```scala
package foo.bar

class Bar extends annotation.StaticAnnotation
```
In a file `Test2.scala`:
```scala
package foo

import foo.bar.Bar

@Bar
class Foo extends annotation.StaticAnnotation
```
In a file `Test3.scala`:
```scala
package object foo {
  @Foo class Hello
}
```
```scala
dotc Test1.scala Test2.scala Test3.scala
-- Warning: Test2.scala:5:1 ----------------------------------------
5 |@Bar
  | ^^^
  | cyclic import foo.bar.Bar, ignored
-- [E006] Unbound Identifier Error: Test2.scala:5:1 ----------------
5 |@Bar
  | ^^^
  | not found: type Bar
```

### PR DESCRIPTION
**Pull request by odersky**
This reverts commit 2b12868070be50fb70c687bcd8a415acbe398e3e.

Since we do not plan to run annotation macros during Namer, there's no need
to annotate before completing. This allows again to define an annotation in the
same scope as an annotated definition. So i3702.scala goes from neg to pos.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/Namer.scala [hunk 1]
```diff
@@ -822,6 +746,22 @@ class Namer { typer: Typer =>
       else completeInCreationContext(denot)
     }
 
+    protected def addAnnotations(sym: Symbol): Unit = original match {
+      case original: untpd.MemberDef =>
+        var hasInlineAnnot = false
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Namer.scala]: Dead variable.

File: compiler/src/dotty/tools/dotc/typer/Namer.scala [hunk 2]
```diff
@@ -822,6 +746,22 @@ class Namer { typer: Typer =>
...
+        lazy val annotCtx =
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Namer.scala]: While you're at it, you may want to make an annotCtx def that can be used both here and for https://github.com/lampepfl/dotty/blob/64bd0fe5210cacd7c17ab285c80712944157c212/compiler/src/dotty/tools/dotc/typer/Typer.scala#L1345 :)

**Comment by allanrenucci** [comment]
Should we also revert #1249?

**Comment by odersky** [comment]
> I think this needs to be implicit ctx => typedAnnotation(annotTree)(annotContext(original, sym))

The last commit caused double definitions of the Annotations class, so the context was too far out. I don't have the time to dig deeper now, so for the moment I am reverting.

**Pull request merged**